### PR TITLE
Add styleUri property in MapInitOptions

### PR DESCRIPTION
**Pull request by Chaoba**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).
Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

## Pull request checklist:
 - [ ] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [x] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog>Add styleUri property in MapInitOptions</changelog>`.

### CODE REVIEW

**Review by kiryldz — APPROVED** [review]

File: sdk/src/main/java/com/mapbox/maps/MapInitOptions.kt [hunk 1]
```diff
@@ -16,6 +16,7 @@ import android.util.AttributeSet
  * @property mapOptions Describes the map options value when using a MapView.
  * @property initialCameraOptions The Initial Camera options when creating a MapView.
  * @property textureView Flag indicating to use a TextureView as render surface for the MapView. Default is false.
+ * @property styleUri The styleUri will applied for the MapView in the onStart lifecycle event if no style is set. Default is [Style.MAPBOX_STREETS].
```
Comment (pengdev) [inline sdk/src/main/java/com/mapbox/maps/MapInitOptions.kt:19]: we should also document that if the field is set to null, no style will be loaded(matches the old behaviour).

File: sdk/src/main/java/com/mapbox/maps/MapInitOptions.kt [hunk 2]
```diff
@@ -24,6 +25,7 @@ data class MapInitOptions constructor(
   var mapOptions: MapOptions = getDefaultMapOptions(context),
   var initialCameraOptions: CameraOptions? = null,
   var textureView: Boolean = false,
+  val styleUri: String? = Style.MAPBOX_STREETS,
```
Comment (pengdev) [inline sdk/src/main/java/com/mapbox/maps/MapInitOptions.kt:28]: Probably we also want to add this field to the xml attributes?

Comment (Chaoba) [reply sdk/src/main/java/com/mapbox/maps/MapInitOptions.kt:28]: Yes, will add it.

File: sdk/src/main/java/com/mapbox/maps/MapView.kt [hunk 3]
```diff
@@ -115,7 +116,8 @@ open class MapView : FrameLayout, MapPluginProviderDelegate, MapControllable {
         MapAttributeParser.parseMapOptions(typedArray, context.resources.displayMetrics.density)
       val cameraOptions = CameraAttributeParser.parseCameraOptions(typedArray)
       val textureView = typedArray.getInt(R.styleable.mapbox_MapView_mapbox_mapSurface, 0) != 0
-      return MapInitOptions(context, resourceOptions, mapOptions, attrs = attrs).also {
+      val styleUri = typedArray.getString(R.styleable.mapbox_MapView_mapbox_styleUri)
```
Comment (pengdev) [inline sdk/src/main/java/com/mapbox/maps/MapView.kt:119]: if the mapbox_MapView_mapbox_styleUri doesn't exit, then the styleUri will be set to null, I think we need to use a default value MAPBOX_STREETS  here, and figure out a way to set an empty StyleUri.

Comment (pengdev) [reply sdk/src/main/java/com/mapbox/maps/MapView.kt:119]: A potential solution would be checking the existence of mapbox_MapView_mapbox_styleUri, if user has specifically set styleUri="" in the xml, we set it to null, if user doesn't specify styleUri in xml, we use default value(MAPBOX_STREETS).

And please also add the usage of this attribute one of our examples to test it out.

Comment (Chaoba) [reply sdk/src/main/java/com/mapbox/maps/MapView.kt:119]: Good catch and solution here 👍

File: app/src/main/java/com/mapbox/maps/testapp/examples/MapboxStudioStyleActivity.kt [hunk 4]
```diff
@@ -13,7 +13,6 @@ class MapboxStudioStyleActivity : AppCompatActivity() {
   override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
     setContentView(R.layout.activity_style_mapbox_studio)
-    mapView.getMapboxMap().loadStyleUri(STUDIO_STYLE_URL)
```
Comment (pengdev) [inline app/src/main/java/com/mapbox/maps/testapp/examples/MapboxStudioStyleActivity.kt:16]: shall we explain the style uri is set through xml here?

File: sdk/src/main/java/com/mapbox/maps/MapInitOptions.kt [hunk 5]
```diff
@@ -24,6 +25,7 @@ data class MapInitOptions constructor(
   var mapOptions: MapOptions = getDefaultMapOptions(context),
   var initialCameraOptions: CameraOptions? = null,
```
Comment (pengdev) [inline sdk/src/main/java/com/mapbox/maps/MapInitOptions.kt:26]: let's rename it to `cameraOptions` to align with iOS team. cc @julianrex

File: sdk/src/main/java/com/mapbox/maps/MapInitOptions.kt (hunk 2 above)
Comment (pengdev) [inline sdk/src/main/java/com/mapbox/maps/MapInitOptions.kt:28]: Let's also change the default value of cacheSize to `1024 * 1024 * 50` to align with iOS. cc @julianrex

File: sdk/src/main/java/com/mapbox/maps/MapInitOptions.kt [hunk 6]
```diff
@@ -39,7 +41,7 @@ data class MapInitOptions constructor(
       .accessToken(CredentialsManager.default.getAccessToken(context))
       .cachePath("${context.filesDir.absolutePath}/$DATABASE_NAME")
       .assetPath(context.filesDir.absolutePath)
-      .cacheSize(50_000_000L) // 50 mb
+      .cacheSize(1024 * 1024 * 50) // 50 mb
```
Comment (pengdev) [inline sdk/src/main/java/com/mapbox/maps/MapInitOptions.kt:44]: We should also change the default value in the `ResourceAttributeParser.kt`

File: sdk/src/main/java/com/mapbox/maps/MapController.kt [hunk 7]
```diff
@@ -111,8 +111,10 @@ internal class MapController : MapPluginProviderDelegate, MapControllable {
     renderer.onStart()
     pluginRegistry.onStart()
     if (!mapboxMap.isStyleLoadInited) {
-      // Load the default style if no style has loaded yet.
-      mapboxMap.loadStyleUri(Style.MAPBOX_STREETS)
+      // Load the style in mapInitOptions if no style has loaded yet.
+      mapInitOptions.styleUri?.let {
+        mapboxMap.loadStyleUri(it)
+      }
```
Comment (pengdev) [inline sdk/src/main/java/com/mapbox/maps/MapController.kt:117]: cc @julianrex to align the behaviour on iOS.

File: sdk/src/main/java/com/mapbox/maps/MapView.kt [hunk 8]
```diff
@@ -115,8 +116,21 @@ open class MapView : FrameLayout, MapPluginProviderDelegate, MapControllable {
         MapAttributeParser.parseMapOptions(typedArray, context.resources.displayMetrics.density)
       val cameraOptions = CameraAttributeParser.parseCameraOptions(typedArray)
       val textureView = typedArray.getInt(R.styleable.mapbox_MapView_mapbox_mapSurface, 0) != 0
-      return MapInitOptions(context, resourceOptions, mapOptions, attrs = attrs).also {
-        it.initialCameraOptions = cameraOptions
+      val styleUri =
+        typedArray.getString(R.styleable.mapbox_MapView_mapbox_styleUri) ?: Style.MAPBOX_STREETS
```
Comment (pengdev) [inline sdk/src/main/java/com/mapbox/maps/MapView.kt:120]: how can user set the  styleUri to null(match the old behaviour)?

File: sdk/src/main/java/com/mapbox/maps/MapController.kt (hunk 7 above)
Comment (pengdev) [reply sdk/src/main/java/com/mapbox/maps/MapController.kt:117]: also note that there's a ticket proposing to rename mapboxMap.iisStyleLoadInited https://github.com/mapbox/mapbox-maps-android-internal/issues/352

File: sdk/src/main/java/com/mapbox/maps/MapView.kt (hunk 8 above)
Comment (Chaoba) [reply sdk/src/main/java/com/mapbox/maps/MapView.kt:120]: User can set this to empty string, we have check in the following line.

Comment (pengdev) [reply sdk/src/main/java/com/mapbox/maps/MapView.kt:120]: oh, sorry didn't notice the check below.

**Review by pengdev — APPROVED** [review]
Great work!

File: sdk/src/main/res/values/attrs.xml [hunk 9]
```diff
@@ -86,5 +86,8 @@
         <attr name="mapbox_cameraPaddingLeft" format="float" />
         <attr name="mapbox_cameraPaddingBottom" format="float" />
         <attr name="mapbox_cameraPaddingRight" format="float" />
+
+        <!-- The default style uri to load when no style is set. -->
+        <attr name="mapbox_styleUri" format="string" />
```
Comment (kiryldz) [inline sdk/src/main/res/values/attrs.xml:91]: Should be added to `sdk/src/main/res-public/values/public.xml` as well.

**Review by kiryldz — CHANGES_REQUESTED** [review]

File: sdk/src/main/res/values/attrs.xml (hunk 9 above)
Comment (pengdev) [reply sdk/src/main/res/values/attrs.xml:91]: good catch

**Review by kiryldz — APPROVED** [review]

**Pull request merged**